language/stdlib/modules/LibraSystem.move

```diff
@@ -154,17 +142,22 @@ module LibraSystem {
         modifies global<LibraConfig::LibraConfig<LibraSystem>>(CoreAddresses::LIBRA_ROOT_ADDRESS());
         include LibraTimestamp::AbortsIfNotOperating;
         include LibraConfig::ReconfigureAbortsIf;
-        aborts_if !ValidatorConfig::spec_is_valid(account_address) with Errors::INVALID_ARGUMENT;
-        aborts_if spec_is_validator(account_address) with Errors::INVALID_ARGUMENT;
-        ensures spec_is_validator(account_address);
+        aborts_if !ValidatorConfig::is_valid(validator_address) with Errors::INVALID_ARGUMENT;
+        aborts_if spec_is_validator(validator_address) with Errors::INVALID_ARGUMENT;
+        /// LIP-6 property: validator has validator role. The code does not check this explicitly,
+        /// but it is implied by the assert ValidatorConfig::is_valid, since
+        /// a published ValidatorConfig has a ValidatorRole is an invariant (in ValidatorConfig).
+        ensures Roles::spec_has_validator_role_addr(validator_address);
```

Comment:
This sounds like the same as the last ensures. What's the difference?

Reply:
If you mean ensures(is_valid....), that implies that it address has a validator role, via another invariant, but doesn't say so directly.  This ensures directly specifies the desired property, and I'm sure it's proved by combining the invariant and is_valid.